aws/resource_aws_db_proxy.go

```diff
@@ -0,0 +1,413 @@
+			userAuthConfig.IAMAuth = aws.String(v)
+		}
+
+		if v, ok := m["secret_arn"].(string); ok && v != "" {
+			userAuthConfig.SecretArn = aws.String(v)
+		}
+
+		if v, ok := m["username"].(string); ok && v != "" {
+			userAuthConfig.UserName = aws.String(v)
+		}
+
+		userAuthConfigs = append(userAuthConfigs, userAuthConfig)
+	}
+
+	return userAuthConfigs
+}
+
+func flattenDbProxyAuth(userAuthConfig *rds.UserAuthConfigInfo) map[string]interface{} {
+	m := make(map[string]interface{})
+
+	m["auth_scheme"] = aws.StringValue(userAuthConfig.AuthScheme)
+	m["description"] = aws.StringValue(userAuthConfig.Description)
+	m["iam_auth"] = aws.StringValue(userAuthConfig.IAMAuth)
+	m["secret_arn"] = aws.StringValue(userAuthConfig.SecretArn)
+	m["username"] = aws.StringValue(userAuthConfig.UserName)
+
+	return m
+}
+
+func flattenDbProxyAuths(userAuthConfigs []*rds.UserAuthConfigInfo) *schema.Set {
+	s := []interface{}{}
+	for _, v := range userAuthConfigs {
+		s = append(s, flattenDbProxyAuth(v))
+	}
+	return schema.NewSet(resourceAwsDbProxyAuthHash, s)
+}
+
+func resourceAwsDbProxyRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	params := rds.DescribeDBProxiesInput{
+		DBProxyName: aws.String(d.Id()),
+	}
+
+	resp, err := conn.DescribeDBProxies(&params)
+	if err != nil {
+		if isAWSErr(err, rds.ErrCodeDBProxyNotFoundFault, "") {
+			log.Printf("[WARN] DB Proxy (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	if len(resp.DBProxies) != 1 ||
+		*resp.DBProxies[0].DBProxyName != d.Id() {
+		return fmt.Errorf("Unable to find DB Proxy: %#v", resp.DBProxies)
+	}
+
+	dbProxy := resp.DBProxies[0]
+
+	d.Set("arn", aws.StringValue(dbProxy.DBProxyArn))
+	d.Set("auth", flattenDbProxyAuths(dbProxy.Auth))
+	d.Set("name", dbProxy.DBProxyName)
+	d.Set("debug_logging", dbProxy.DebugLogging)
+	d.Set("engine_family", dbProxy.EngineFamily)
+	d.Set("idle_client_timeout", dbProxy.IdleClientTimeout)
+	d.Set("require_tls", dbProxy.RequireTLS)
+	d.Set("role_arn", dbProxy.RoleArn)
+	d.Set("vpc_subnet_ids", flattenStringSet(dbProxy.VpcSubnetIds))
+	d.Set("security_group_ids", flattenStringSet(dbProxy.VpcSecurityGroupIds))
+
+	tags, err := keyvaluetags.RdsListTags(conn, d.Get("arn").(string))
+
+	if err != nil {
+		return fmt.Errorf("Error listing tags for RDS DB Proxy (%s): %s", d.Get("arn").(string), err)
+	}
+
+	// if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+	// 	return fmt.Errorf("error setting tags: %s", err)
+	// }
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("Error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsDbProxyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	oName, nName := d.GetChange("name")
```

Comment:
can you wrap the `ModifyDBProxyInput` with some kind of `d.HasChange/s` to prevent updating the proxy when only the tags need updating?

Reply:
This change does need to happen to prevent the extra API call, e.g.

```go
if d.HasChanges(
	"auth",
	"debug_logging",
	"idle_client_timeout",
	"name",
	"require_tls",
	"role_arn",
	"vpc_security_group_ids") {
// ... existing logic up to d.HasChange("tags")
}

if d.HasChange("tags") {
// ...
```